conan/tools/premake/premake.py

```diff
@@ -0,0 +1,39 @@
+class Premake(object):
+    def __init__(self, conanfile):
+        self._conanfile = conanfile
+
+    # automatically chooses premake action based on used compiler
+    def configure(self):
+        if "Visual Studio" in self.settings.compiler:
+            _visuals = {'8': '2005',
+                        '9': '2008',
+                        '10': '2010',
+                        '11': '2012',
+                        '12': '2013',
+                        '14': '2015',
+                        '15': '2017',
+                        '16': '2019'}
```

Comment:
Many thanks for contributing this PR. It has been merged, it will be in next 1.44.
If you keep using it, please try to add VS 2022 support as soon as you upgrade to it, there are users already using VS2022 with other generators.

Reply:
according to [Premake's documentation](https://premake.github.io/docs/Using-Premake) there's no VS 2022 generator yet.

Reply:
Oh, I see. Good, then users will need to wait to Premake to add it, lets wait then.